functions/Get-DbaInstallDate.ps1

```diff
@@ -0,0 +1,188 @@
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>
+
+.LINK
+https://dbatools.io/Get-DbaInstallDate
+
+.EXAMPLE
+Get-DbaInstallDate -SqlServer SqlBox1\Instance2
+
+Returns an object with SQL Server Install date as a string and the Windows install date as string. 
+
+.EXAMPLE
+Get-DbaInstallDate -SqlServer winserver\sqlexpress, sql2016
+
+Returns an object with SQL Server Install date as a string and the Windows install date as a string for both SQLInstances that are passed to the cmdlet.  
+	
+.EXAMPLE   
+Get-DbaInstallDate -SqlServer sqlserver2014a, sql2016 -SqlOnly
+
+Returns an object with only the SQL Server Install date as a string. 
+
+.EXAMPLE   
+Get-DbaInstallDate -SqlServer sqlserver2014a, sql2016 -WindowsOnly
+
+Returns an object with only the Windows Install date as a string. 
+
+.EXAMPLE   
+Get-SqlRegisteredServerName -SqlServer sql2014 | Get-DbaInstallDate
+
+Returns an object with SQL Server Install date as a string and the Windows install date as string for every server listed in the Central Management Server on sql2014
+	
+#>
+	[CmdletBinding(DefaultParameterSetName = "Default")]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlInstance", "ComputerName")]
+		[object[]]$SqlServer,
+		[parameter(ParameterSetName = "Sql")]
+		[Switch]$SqlOnly,
+		[parameter(ParameterSetName = "Windows")]
+		[Switch]$WindowsOnly,
+		[Alias("Credential")]
+		[PsCredential]$SqlCredential,
+		[PsCredential]$WindowsCredential
+	)
+	
+	PROCESS
+	{
+		foreach ($instance in $SqlServer)
+		{
+			if ($instance.Gettype().FullName -eq [System.Management.Automation.PSCustomObject] )
+			{
+				$servername = $instance.SqlInstance
+			}
+			elseif ($instance.Gettype().FullName -eq [Microsoft.SqlServer.Management.Smo.Server])
+			{
+				$servername = $instance.NetName
+			}
+			else
+			{
+				$servername = $instance
+			}
+						
+			if ($WindowsOnly -ne $true)
+			{ 
+				try 
+				{
+					Write-Message -Level Verbose -Message "Connecting to $instance" -Silent $false 
+                    if ($SqlCredential)
+                    { 
+                        $server = Connect-SqlInstance -SqlInstance $servername -SqlCredential $SqlCredential #-ErrorVariable ConnectError
+                    } else 
+                    { 
+                        $server = Connect-SqlInstance -SqlInstance $servername 
+                    }					
+				}
+				catch 
+				{
+					Stop-Function -Message "Failed to connect to: $instance" -Continue -Target $instance					
+				}
+
+				if ( $server.VersionMajor -ge 9 )
+				{ 
+					Write-Message -Level Verbose -Message "Getting Install Date for: $instance" -Silent $false 
+					$sql = "SELECT create_date FROM sys.server_principals WHERE sid = 0x010100000000000512000000"
+					$sqlInstallDate = $server.databases['master'].ExecuteWithResults($sql).tables
```

Comment:
PR #1365 was added we can now declare this as `$server.Query($db,$sql).Tables`. Also a good idea to use `Tables[0]`, although this should only return one dataset it is just on the safe side.